src/commands/login.rs

```diff
@@ -367,6 +372,18 @@ impl LoginCommand {
         }
     }
 
+    fn validate_hippo_url(&self) -> Result<(), anyhow::Error> {
+        let url = self.url();
+        match Url::parse(url) {
+            Ok(_) => Result::Ok(()),
+            Err(error) => Result::Err(anyhow::format_err!(
+                "Invalid Hippo URL, should be in the format \"https://my-hippo-instance.com\". \"{}\": {}",
+                url,
+                error
+            )),
+        }
```

Comment:
There are a couple of nicer options here, depending on exactly what you want the errors to look like.

The easiest option (which we use all over the place) is the `.with_context` method, which attaches context to another error, e.g.:

```suggestion
        Url::parse(url).with_context(|| {
            format!("Invalid Hippo URL, should be in the format \"https://my-hippo-instance.com\". {url:?}")
        })
```

That will give you a chained "caused by" list of errors, which might not be what you want here.

A shorter way to get (almost) exactly the error above (untested, hopefully right):

```suggestion
        Url::parse(url)
            .map_err(|err| anyhow::anyhow!(
                "Invalid Hippo URL, should be in the format \"https://my-hippo-instance.com\". {url:?}: {err}"
            )
```